compiler/main/checks.cpp

```diff
@@ -191,6 +243,35 @@ void check_cullOverReferences()
   for_alive_in_Vec(ContextCallExpr, cc, gContextCallExprs) {
     INT_FATAL("ContextCallExpr should no longer be in AST");
   }
+
+  // for all CallExprs, if we call a promotion wrapper that is marked no promotion, warn
+  for_alive_in_Vec(CallExpr, ce, gCallExprs) {
+    if (FnSymbol* fn = ce->theFnSymbol()) {
+      if (fn->hasFlag(FLAG_PROMOTION_WRAPPER) &&
+          fn->hasFlag(FLAG_NO_PROMOTION_WHEN_BY_REF)) {
+
+        // We cannot rely on retTag to tell us if this promoted function returns
```

Comment:
> We cannot rely on retTag to tell us if this promoted function returns a ref or not

Is the idea if the result of the "promotion wrapper" function goes into a ref than we know its something we're assigning into?

So if I have `A = B[C]` then I'd generate a wrapper generated for indexing through `B`, and I guess I store the results of that maybe into some compiler-generate temporary? But then that temporary is never used as a ref?

Reply:
If you have `A = B[C]` where `A` is a `ref` would be invalid code due to lifetime checking. If `A` is  a `var`, its just a copy.

The problem is that while creating and resolving the promotion wrapper, we wipe out the `retTag` info and just always set it to `RET_VALUE`.